payments-core/src/main/java/com/stripe/android/IntentConfirmationInterceptor.kt

```diff
@@ -0,0 +1,226 @@
+package com.stripe.android
+
+import androidx.annotation.RestrictTo
+import com.stripe.android.core.injection.PUBLISHABLE_KEY
+import com.stripe.android.core.injection.STRIPE_ACCOUNT_ID
+import com.stripe.android.core.networking.ApiRequest
+import com.stripe.android.model.ConfirmPaymentIntentParams
+import com.stripe.android.model.ConfirmPaymentIntentParams.SetupFutureUsage.OffSession
+import com.stripe.android.model.ConfirmStripeIntentParams
+import com.stripe.android.model.PaymentMethod
+import com.stripe.android.model.PaymentMethodCreateParams
+import com.stripe.android.model.StripeIntent
+import com.stripe.android.networking.StripeRepository
+import javax.inject.Inject
+import javax.inject.Named
+
+@RestrictTo(RestrictTo.Scope.LIBRARY_GROUP)
+interface IntentConfirmationInterceptor {
+
+    @RestrictTo(RestrictTo.Scope.LIBRARY_GROUP)
+    sealed interface NextStep {
+
+        @RestrictTo(RestrictTo.Scope.LIBRARY_GROUP)
+        data class Fail(val error: Throwable) : NextStep
+
+        @RestrictTo(RestrictTo.Scope.LIBRARY_GROUP)
+        data class Confirm(val confirmParams: ConfirmStripeIntentParams) : NextStep
+
+        @RestrictTo(RestrictTo.Scope.LIBRARY_GROUP)
+        data class HandleNextAction(val clientSecret: String) : NextStep
+
+        @RestrictTo(RestrictTo.Scope.LIBRARY_GROUP)
+        object Complete : NextStep
+    }
+
+    suspend fun intercept(
```

Comment:
We should be adding docs for these, even if they're WIP. It's unclear why we need 2 intercept methods.